### DISCUSSION TITLE
Is it possible to disable auto-loading "Foreign Key" as a model

### DISCUSSION THREAD

**Discussion by bilyalov** [collerek/ormar#284] [Q&A]
I have a model:
```
class Post(ormar.Model):
    id: int = ormar.Integer(primary_key=True)
    title: str = ormar.String(max_length=500)
    link: PageLink = ormar.ForeignKey(PageLink, related_name='posts')
```

When I make a request to get a list of posts (`Post.objects.all()`), I want to get a list of posts where the link field is a numeric value, while having all the ormar methods for working with the Post model
But I get a list of posts where the link field is an instance of the PageLink model with the pk field. It looks like this
```
{
    "id": 1,
    "title": "my post",
    "link": {
        "id": 1"
    }
}
```

![image](https://user-images.githubusercontent.com/25269937/126777080-b3aa7706-2ebc-403d-85f0-77a4475b9485.png)

Is it possible to disable this behavior?

**Comment by collerek** [comment]
As of now - no.

The reason is that when you would replace it with the value you wouldn't be able to load the model of needed anymore and loose all Relation specific attributes.

You would like to disable it for one query, always? What is your use case?

**Reply by bilyalov** [reply]
Yes, it would be convenient to disable only on a specific request. And load the field as a model if necessary, for example, using the "load" method.

Or add `<field>_id` field that will contain the key value

In any case, it's up to you :) Now I have found a way to do without this, and it works well. Thank you for working on this project!

**Reply by collerek** [reply]
If you want to just get the data as dict or tuple you can use `values` and `values_list`, that way the related model won't be loaded (but the main one won't be loaded too 😅 ). I know it's not perfect in your use case, but as a workaround it might work :)

**Reply by artel1992** [reply]
> Yes, it would be convenient to disable only on a specific request. And load the field as a model if necessary, for example, using the "load" method.
> 
> Or add `<field>_id` field that will contain the key value
> 
> In any case, it's up to you :) Now I have found a way to do without this, and it works well. Thank you for working on this project!

How do you do it? I need have only integer value too
